lib/web/apiserver.go

```diff
@@ -2403,6 +2403,40 @@ func (h *Handler) WithRedirect(fn redirectHandlerFunc) httprouter.Handle {
 	}
 }
 
+// WithJavascriptRedirect is a handler that redirects to the path specified
+// using Javascript rather than HTTP. This is needed for redirects that can
+// have a header size larger than 8kb, which some middlewares will drop.
+// See https://github.com/gravitational/teleport/issues/7467.
+func (h *Handler) WithJavascriptRedirect(fn redirectHandlerFunc) httprouter.Handle {
+	return func(w http.ResponseWriter, r *http.Request, p httprouter.Params) {
+		httplib.SetNoCacheHeaders(w.Header())
+		nonce, err := utils.CryptoRandomHex(auth.TokenLenBytes)
+		if err != nil {
```

Comment:
To be on the safe side I suggest to add some basic checks. Something like below (just to give an idea)

```golang
	if !strings.HasPrefix(redirect, "https://"){
		return trace.Wrap("invalid redirect URL")
	}

	// to ensure a valid JS string 
	// (or do URI encoding instead)
	bytes, err := json.Marshal(struct {
		Redirect string `json:"redirect"`
	}{
		Redirect: value,
	})

	fmt.Fprintf(w, "var teleport = %v;\n", string(bytes))
```